File: easy/1984_minimum_difference_between_highest_and_lowest_of_k_scores/solution.py
from typing import List


class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()

        min_diff = 100000

        for i in range(len(nums) - k + 1):
            min_diff = min(min_diff, nums[i + k - 1] - nums[i])

        return min_diff

File: easy/1984_minimum_difference_between_highest_and_lowest_of_k_scores/solution.cpp
#include <vector>

class Solution {
public:
  int minimumDifference(std::vector<int> &nums, int k) {
    sort(nums.begin(), nums.end());

    int minDiff = 100000;

    for (int i = 0; i < nums.size() - k + 1; ++i)
      minDiff = std::min(minDiff, nums[i + k - 1] - nums[i]);

    return minDiff;
  }
};
